yarn-project/aztec-rpc/src/database/memory_db.ts

```diff
@@ -19,11 +19,31 @@ export class MemoryDB extends MemoryContractDatabase implements Database {
   private treeRoots: Record<MerkleTreeId, Fr> | undefined;
   private globalVariablesHash: Fr | undefined;
   private addresses: CompleteAddress[] = [];
+  private eip1271Witnesses: Record<string, Fr[]> = {};
 
   constructor(logSuffix?: string) {
     super(createDebugLogger(logSuffix ? 'aztec:memory_db_' + logSuffix : 'aztec:memory_db'));
   }
 
+  /**
+   * Add a eip1271 witness to the database.
+   * @param messageHash - The message hash.
+   * @param witness - An array of field elements representing the eip1271 witness.
+   */
+  public addEip1271Witness(messageHash: Fr, witness: Fr[]): Promise<void> {
+    this.eip1271Witnesses[messageHash.toString()] = witness;
+    return Promise.resolve();
+  }
+
+  /**
+   * Fetching the eip1271 witness for a given message hash.
+   * @param messageHash - The message hash.
+   * @returns A Promise that resolves to an array of field elements representing the eip1271 witness.
+   */
+  public getEip1271Witness(messageHash: Fr): Promise<Fr[]> {
+    return Promise.resolve(this.eip1271Witnesses[messageHash.toString()]);
```

Comment:
What should we return if there's no value for this key? If it's undefined (what I'd go with) we need to adjust the return type here.